src/nspawn/nspawn.c

```diff
@@ -900,13 +900,12 @@ static int parse_argv(int argc, char *argv[]) {
 
                 case ARG_PRIVATE_USERS:
 
-                        r = optarg ? parse_boolean(optarg) : 1;
-                        if (r == 0) {
+                        if (streq_ptr(optarg, "no")) {
                                 /* no: User namespacing off */
                                 arg_userns_mode = USER_NAMESPACE_NO;
                                 arg_uid_shift = UID_INVALID;
                                 arg_uid_range = UINT32_C(0x10000);
-                        } else if (r > 0) {
+                        } else if (!optarg || streq(optarg, "yes")) {
```

Comment:
We do this "extended" boolean logic all the time, I think we should do the same here. i.e. first parse the parameter as something other, and if that fails as true boolean with our usual parse_boolean() call... I'd agree with @evverx, we should also be able to use --private-users=true here...